src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -399,14 +399,14 @@ private function describeValue($value, $omitTags, $showArguments)
             );
         }
 
-        if ($value instanceof Definition) {
-            return $this->getContainerDefinitionData($value, $omitTags, $showArguments);
-        }
-
         if ($value instanceof ArgumentInterface) {
             return $this->describeValue($value->getValues(), $omitTags, $showArguments);
         }
 
+        if ($value instanceof Definition) {
```

Comment:
@nicolas-grekas existing bundles may have compiler passes processing the container. And Symfony 3.3 will change some existing services to use such argument.
Pure ArgumentInterface would be an argument for which they cannot do special processing. A Definition which must not be processed like a Definition is much worse, as they will *think* they can process it.